packages/app/e2e/app.v9.e2e.js

```diff
@@ -0,0 +1,316 @@
+      const { initializeApp, getApps, getApp } = modular;
+
+      const appCount = firebase.apps.length;
+      const name = `testscoreapp${FirebaseHelpers.id}`;
+      const platformAppConfig = FirebaseHelpers.app.config();
+      const newApp = await initializeApp(platformAppConfig, name);
+      newApp.name.should.equal(name);
+      newApp.toString().should.equal(name);
+      newApp.options.apiKey.should.equal(platformAppConfig.apiKey);
+
+      const apps = getApps();
+
+      should.equal(apps.includes(getApp(name)), true);
+      should.equal(apps.length, appCount + 1);
+      return newApp.delete();
+    });
+
+    it('should error if dynamic app initialization values are incorrect', async function () {
+      const { initializeApp, getApps } = modular;
+
+      const appCount = getApps().length;
+      try {
+        await initializeApp({ appId: 'myid' }, 'myname');
+        throw new Error('Should have rejected incorrect initializeApp input');
+      } catch (e) {
+        e.message.should.equal("Missing or invalid FirebaseOptions property 'apiKey'.");
+        should.equal(getApps().length, appCount);
+        should.equal(getApps().includes('myname'), false);
+      }
+    });
+
+    it('should error if dynamic app initialization values are invalid', async function () {
+      const { initializeApp, getApps } = modular;
+
+      // firebase-android-sdk will not complain on invalid initialization values, iOS throws
+      if (device.getPlatform() === 'android') {
+        return;
+      }
+
+      const appCount = getApps().length;
+      try {
+        const firebaseConfig = {
+          apiKey: 'XXXXXXXXXXXXXXXXXXXXXXX',
+          authDomain: 'test-XXXXX.firebaseapp.com',
+          databaseURL: 'https://test-XXXXXX.firebaseio.com',
+          projectId: 'test-XXXXX',
+          storageBucket: 'tes-XXXXX.appspot.com',
+          messagingSenderId: 'XXXXXXXXXXXXX',
+          appId: '1:XXXXXXXXX',
+          app_name: 'TEST',
+        };
+        await initializeApp(firebaseConfig, 'myname');
+        throw new Error('Should have rejected incorrect initializeApp input');
+      } catch (e) {
+        e.code.should.containEql('app/unknown');
+        e.message.should.containEql('Configuration fails');
+        should.equal(firebase.apps.length, appCount);
+        should.equal(firebase.apps.includes('myname'), false);
+      }
+    });
+
+    it('apps can be deleted, but only once', async function () {
+      const { initializeApp, getApp, deleteApp } = modular;
+
+      const name = `testscoreapp${FirebaseHelpers.id}`;
+      const platformAppConfig = FirebaseHelpers.app.config();
+      const newApp = await initializeApp(platformAppConfig, name);
+
+      newApp.name.should.equal(name);
+      newApp.toString().should.equal(name);
+      newApp.options.apiKey.should.equal(platformAppConfig.apiKey);
+
+      await deleteApp(newApp);
+      try {
+        await deleteApp(newApp);
+      } catch (e) {
```

Comment:
this one and the next five that should throw either need to have a fail after the line that throws (like a couple cases above) to make *sure* it threw, or they should be some special assertion that expects an exception.

I think we tried the method that was like `expectException` or whatever they called it and it wasn't working for some unimportant reason so we typically just throw something like 
```javascript
        throw new Error('Should have rejected incorrect initializeApp input');
````
after the lines that should throw

Reply:
Oh okay, I see why we need to do that, thanks :) 
I've added this to the following cases :)